data/data/gcp/network/external-lb.tf

```diff
@@ -0,0 +1,17 @@
+resource "google_compute_address" "cluster_public_ip" {
+  name = "${var.cluster_id}-cluster-public-ip"
+}
+
+resource "google_compute_target_pool" "api_external" {
+  name = "${var.cluster_id}-api-external"
+
+  instances = concat(var.master_instances, [var.bootstrap_instance])
```

Comment:
were you planning on adding a health check to this target pool?

Reply:
It looks like terraform provider will only allow legacy google_compute_http_health_check here. I don't know if we have access to non https ports to test. Thoughts?

https://www.terraform.io/docs/providers/google/r/compute_target_pool.html#health_checks

Reply:
Yup only the legacy health checks are allowed on network load balancers, but you should be able to specify the port (https://www.terraform.io/docs/providers/google/r/compute_http_health_check.html#port) and afaict it does check either http or https

Reply:
also we need to make sure we have firewall rules:
internal: https://cloud.google.com/load-balancing/docs/health-checks#fw-rule
external: https://cloud.google.com/load-balancing/docs/health-checks#fw-netlb

Reply:
@csrwng I added http health checks. We will have to see how they work when we get functioning cluster(s).

Reply:
@abhinavdahiya The internal lb should fall into the network_cidr range and be handled by existing rules. I added one for the external lb.

Reply:
hmm, the internal network load balancer requires `35.191.0.0/16` and `130.211.0.0/22` ranges to be allowed to talk to the control-plane...

Reply:
https://cloud.google.com/load-balancing/docs/health-check-concepts#multiple-probers
```
GCP sends health check probes from multiple redundant systems from the appropriate source IP ranges. No single probe system is responsible for all of the probes. Multiple systems issue probes simultaneously so that failure of one does not cause GCP to lose track of VM health states.
```